src/core/optimization_container.jl

```diff
@@ -583,7 +583,7 @@ function _add_variable_container!(
     axs...,
 ) where {T <: VariableType, U <: Union{PSY.Component, PSY.System}}
     if sparse
-        var_container = sparse_container_spec(JuMP.VariableRef, axs...)
+        var_container = sparse_container_spec(Float64, axs...)
```

Comment:
```suggestion
       # We initialize sparse containers with Floats64, not ideal and introduces type instability, because JuMP.Containers.SparseAxisArrays can't be initialized with undef
        var_container = sparse_container_spec(Float64, axs...)
```